pkg/util/ecs/metadata/clients.go

```diff
@@ -122,7 +123,7 @@ func newAutodetectedClientV1() (*v1.Client, error) {
 // newClientV3ForContainer detects the metadata API v3 endpoint for the specified
 // container and creates a new client for it.
 func newClientV3ForContainer(id string) (*v3.Client, error) {
-	agentURL, err := getAgentV3URLFromDocker(id)
+	agentURL, err := getAgentV3URLFromDocker(context.TODO(), id)
```

Comment:
a `context.Context` can be add to the `newClientV3ForContainer` function.

Reply:
Done: https://github.com/DataDog/datadog-agent/pull/7605/commits/c5781ca9257096a9c3394369ade30745b9b68a03